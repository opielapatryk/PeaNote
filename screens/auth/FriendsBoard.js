import { Text,TextInput,View,Button } from 'react-native'
import React,{useState} from 'react'
import axios from 'axios'
import * as SecureStore from 'expo-secure-store';
// import NoteCreateInput from '../../components/NoteCreateInput';
// import { Note } from '../../components/Note';
// import { useDispatch, useSelector} from 'react-redux';
// import { styles } from '../../assets/styles';
// import {changeInfo} from '../../store/boardSlice';

export default FriendsBoard = ({ route,navigation }) => {
    const { friendId, friendName } = route.params;
    const [content,setContent] = useState('');

    const createNote = async () => {
        try {
            currentUserId = await SecureStore.getItemAsync('userId');
            const userURL = `http://localhost:8000/api/users/${currentUserId}/`

            const result = await axios.post(`http://localhost:8000/api/stickers/`,{
                'content':content,
                'creator':userURL
            })

            const stickerID = result.data.id

            const friendURL = `http://localhost:8000/api/users/${friendId}/`

            const resultStickers = await axios.get(friendURL)
    
            let list = resultStickers.data.stickersOnBoard
    
            console.log(list);
    
            list.push(`http://localhost:8000/api/stickers/${stickerID}/`);
    
            console.log(list);

            console.log('friend ID: ' + friendId);
            await axios.patch(`http://localhost:8000/api/users/${friendId}/`,{
                'stickersOnBoard': list
            })
        } catch (error) {
            console.log(error.message);
        }
    }

    const removeFriend = async () => {
        try {
            currentUserId = await SecureStore.getItemAsync('userId');
            const currentUserResult = await axios.get(`http://localhost:8000/api/users/${currentUserId}`)

            let list = currentUserResult.data.friends

            list = list.filter((element) => element !== `http://localhost:8000/api/users/${friendId}/`)

            console.log(list);

            const resp = await axios.patch(`http://localhost:8000/api/users/${currentUserId}/`,{
                'friends':list
            })

            console.log(resp.status);

            if(resp.status === 200) {
                navigation.navigate('Friends')
            }
            return resp
        } catch (error) {
            console.log(error.message);
        }
    }

    return (
      <View>
        <Button title='remove friend' onPress={()=>removeFriend()}/>
        <Text>Welcome on {friendName?friendName:'your friend'} board</Text>
        <TextInput placeholder='note.....' value={content} onChangeText={(content)=>setContent(content)}/>
        <Button title='create note' onPress={()=>createNote()}/>
      </View>
    );
}