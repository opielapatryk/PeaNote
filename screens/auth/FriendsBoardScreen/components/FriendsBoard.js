import React, { useState } from 'react';
import { Text, TextInput, View, Pressable, Keyboard,  TouchableWithoutFeedback } from 'react-native';
import {createNote} from '../functions/createNote'
import {removeFriend} from '../functions/removeFriend'
import {styles} from '../../../../assets/styles/styles'
import { useDispatch } from 'react-redux';

const FriendsBoard = ({ route, navigation }) => {
  const { friendEmail } = route.params;
  const [content, setContent] = useState('');
  const [message, setMessage] = useState('');
  const dispatch = useDispatch()

  return (
    <TouchableWithoutFeedback 
    onPress={() => Keyboard.dismiss()}>
      <View style={styles.friendsboard}>
        <View>
          <TextInput style={styles.friendsTextInput} placeholder={message?message:"NEW NOTE"} value={content} onChangeText={text=>setContent(text)} autoCapitalize="sentences"
          autoCorrect={false} maxLength={100} multiline/>

          <Pressable style={styles.friendsHeaderRequest} onPress={()=>createNote(content,setContent,setMessage,friendEmail)}><Text style={styles.removeFriendText}>CREATE NOTE</Text></Pressable>
          
        </View>


        <Pressable style={styles.deleteAccountButton} onPress={()=>removeFriend(navigation,friendEmail,dispatch)}><Text style={styles.deleteAccountText}>REMOVE FRIEND</Text></Pressable>
      </View>
    </TouchableWithoutFeedback>
  );
};

export default FriendsBoard;
