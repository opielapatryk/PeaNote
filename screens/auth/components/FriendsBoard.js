import React, { useState } from 'react';
import { Text, TextInput, View, Button } from 'react-native';
import {createNote,removeFriend} from '../logic/apiFriendsBoard'

const FriendsBoard = ({ route, navigation }) => {
  const { friendId, friendName } = route.params;
  const [content, setContent] = useState('');
  const [message, setMessage] = useState('');

  return (
    <View>
      <Button title="Remove Friend" onPress={()=>removeFriend(navigation,friendId)} />
      <Text>Welcome to {friendName ? friendName + "'s" : 'your'} board</Text>
      <TextInput placeholder="Note....." value={content} onChangeText={(content)=>setContent(content)} />
      <Button title="Create Note" onPress={()=>createNote(content,setContent,setMessage,friendId)} />
      <Text>{message}</Text>
    </View>
  );
};

export default FriendsBoard;
