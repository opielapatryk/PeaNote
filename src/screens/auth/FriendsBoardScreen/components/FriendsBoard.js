import React, { useState } from 'react';
import { Text, TextInput, View, Pressable, Keyboard,  TouchableWithoutFeedback,Image } from 'react-native';
import {createNote} from '../functions/createNote'
import {removeFriend} from '../functions/removeFriend'
import {styles} from '../../../../../assets/styles/styles'
import { useDispatch,useSelector } from 'react-redux';
import { setFriendimage } from '../../../../store/settings/settingsSlice';
import auth from '@react-native-firebase/auth';
import firestore from '@react-native-firebase/firestore';
import { useFocusEffect } from '@react-navigation/native';
import {ensureDirExists} from "../../BoardScreen/functions/ensureDirExists";
import {getSingleImg} from '../../BoardScreen/functions/getSingleImg';

const FriendsBoard = ({ route, navigation }) => {
  const [description, newDescription] = useState(reduxdescription);
  const { friendEmail,name,oldnickname} = route.params;
  const [content, setContent] = useState('');
  const [message, setMessage] = useState('');
  const dispatch = useDispatch()
  const { friendimage } = useSelector((state) => state.settings);
  const { reduxdescription } = useSelector((state) => state.login);

  const EMAIL = auth().currentUser.email

  const downloadImage = async () => {
    await ensureDirExists();

    const fileUri = await getSingleImg(friendEmail)

    dispatch(setFriendimage(fileUri));
  }

  useFocusEffect(
    React.useCallback(() => {
      const getEmailByUsername = async () =>{
        const usernameSnapshot = await firestore()
          .collection('users')
          .where('email', '==', friendEmail)
          .get();
          
          newDescription(usernameSnapshot.docs[0].data().description)
      }

      getEmailByUsername()

      downloadImage()
      
    }, [])
  );


  const [showInput,setShowInput] = useState(false)
  const [nickname,setNewNickName] = useState('')
  const [nicknameMessage,setNewNickNameMessage] = useState('')
  
  const handleNickNameChange = async () => {
    if (showInput) {

      const getUserByEmail = await firestore()
        .collection('users')
        .where('email', '==', EMAIL)
        .get()

      getUserByEmail.forEach((doc)=>{
        let friends = doc.data().friends

        friends.forEach((friend)=>{
          if(friend.email === friendEmail){
            
            friend['nickname'] = nickname

          }
        });

        firestore()
          .collection('users')
          .doc(doc.id)
          .update({
            friends: friends,
          }).then(()=>setNewNickNameMessage('nickname changed')).catch(()=>setNewNickNameMessage('try different nickname'))

      })

      setNewNickName('')
      
      
      setTimeout(() => {
        setShowInput(false)
        setNewNickNameMessage('')
      }, 2000);
    } else {
      setShowInput(true)
    }
  };

  return (
    <TouchableWithoutFeedback 
    onPress={() => Keyboard.dismiss()}>
      <View style={{flex: 1,
    backgroundColor: '#FFF',
    justifyContent:"space-between",}}>
      <View style={{alignItems:'center',backgroundColor:'white'}}>
        <View style={styles.ProfilePicParent}>
        {friendimage && <Image source={{uri: friendimage}} style={styles.ProfilePic}/>}
        </View>

        <View style={[styles.friendsHeaderRequest,{height:50}]}>
          <Text style={{textAlign:'center',fontStyle:'italic'}}>{description}</Text>
        </View>
        

        <TextInput style={[styles.friendsTextInput,{paddingTop:10}]} placeholder={message?message:"NEW NOTE"} value={content} onChangeText={text=>setContent(text)} autoCapitalize="sentences"
          autoCorrect={false} maxLength={100} multiline/>

          <Pressable style={styles.friendsHeaderRequest} onPress={()=>createNote(content,setContent,setMessage,friendEmail,name)}><Text style={styles.removeFriendText}>CREATE NOTE</Text></Pressable>


          {showInput && (
          <TextInput
            style={styles.friendsTextInput}
            placeholder={nicknameMessage?nicknameMessage:"NICKNAME"}
            onChangeText={text=>setNewNickName(text)}
            value={nickname}
            maxLength={25}
          />   
        )}

        <Pressable style={styles.friendsHeaderRequest} onPress={handleNickNameChange}>
          <Text style={styles.removeFriendText}>{showInput ? 'CONFIRM' : 'SET NICKNAME'}</Text>
        </Pressable>
        
      </View>

      <View>
        <Text style={[styles.removeFriendText,{fontSize:14,marginBottom:10}]}>{oldnickname?friendEmail:''}</Text>
        
        <Pressable style={styles.deleteAccountButton} onPress={()=>removeFriend(navigation,friendEmail,name,nickname,dispatch)}><Text style={styles.removeFriendText}>REMOVE FRIEND</Text></Pressable>
      </View>
      
      </View>
    </TouchableWithoutFeedback>
  );
};

export default FriendsBoard;
