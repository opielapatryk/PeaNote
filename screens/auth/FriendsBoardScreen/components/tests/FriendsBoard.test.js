import React from 'react';
import renderer from 'react-test-renderer';
import FriendsBoard from '../FriendsBoard';

jest.mock('@react-native-firebase/auth', () => () => {
    return {
        currentUser: {
            email:'test@example.com'
        }
    }
  })

jest.mock('@react-native-firebase/firestore', () => ({
    collection: jest.fn(),
    doc: jest.fn(),
    get: jest.fn(),
}));

test('FriendsBoard screen renders correctly.', async () => {
    const route = {
      params: {
        friendEmail: 'test@example.com',
      },
    };
    const navigation = {
      navigate: jest.fn(),
    };

    const tree = renderer.create(
        <FriendsBoard route={route} navigation={navigation} />
      ).toJSON();
    expect(tree).toMatchSnapshot();
})
