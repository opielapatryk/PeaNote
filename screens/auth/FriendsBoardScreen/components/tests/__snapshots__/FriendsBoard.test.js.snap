// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FriendsBoard screen renders correctly. 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#FFFDF3",
      "flex": 1,
      "justifyContent": "space-between",
    }
  }
>
  <View>
    <TextInput
      autoCapitalize="sentences"
      autoCorrect={false}
      maxLength={100}
      multiline={true}
      onChangeText={[Function]}
      placeholder="NEW NOTE"
      style={
        {
          "backgroundColor": "#FFF",
          "fontSize": 17,
          "fontWeight": "bold",
          "height": 71.72043010752688,
          "letterSpacing": 0.5,
          "paddingTop": 10,
          "shadowOffset": {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.18,
          "shadowRadius": 1,
          "textAlign": "center",
          "width": 750,
        }
      }
      value=""
    />
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "#FFF",
          "height": 72.5,
          "justifyContent": "center",
          "shadowOffset": {
            "height": 0.5,
            "width": 0,
          },
          "shadowOpacity": 0.1,
          "shadowRadius": 0.5,
          "width": 750,
          "zIndex": 1000,
        }
      }
    >
      <Text
        style={
          {
            "alignSelf": "center",
            "elevation": 1,
            "fontSize": 17,
            "fontWeight": "bold",
            "letterSpacing": 1,
            "shadowOffset": {
              "height": 1,
              "width": 0,
            },
            "shadowOpacity": 0.18,
            "shadowRadius": 1,
          }
        }
      >
        CREATE NOTE
      </Text>
    </View>
    <Text
      style={
        {
          "alignSelf": "center",
          "fontStyle": "italic",
          "margin": 10,
        }
      }
    />
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#FFF",
        "height": 107.58064516129032,
        "justifyContent": "center",
        "paddingBottom": 13.34,
        "shadowOffset": {
          "height": 0.5,
          "width": 0,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 0.5,
        "width": 750,
        "zIndex": 1000,
      }
    }
  >
    <Text
      style={
        {
          "alignSelf": "center",
          "elevation": 1,
          "fontSize": 17,
          "fontWeight": "bold",
          "letterSpacing": 2,
          "shadowOffset": {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.18,
          "shadowRadius": 1,
        }
      }
    >
      REMOVE FRIEND
    </Text>
  </View>
</View>
`;
